debugger/src/debug_adapter/dap_adapter.rs

```diff
@@ -111,6 +118,61 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
         }
     }
 
+    /// Handles the `terminate` request, and very specifically only terminates the target appliction,
+    /// and does not impact the debug sessions status.
+    pub(crate) fn terminate(
+        &mut self,
+        target_core: &mut CoreHandle,
+        request: Request,
+    ) -> Result<()> {
+        // TODO: For now (until we do multicore), we will assume that terminate request translates to a halt of the core.
+        match target_core.core.halt(Duration::from_millis(500)) {
+            Ok(_) => self.send_response::<TerminateResponse>(request, Ok(None)),
+            // An error here is not fatal, we will just send a response and let the DAP client force a disconnect.
+            Err(error) => {
+                let debug_error = format!("Failed to halt the target: {error:?}");
+                self.send_response::<TerminateResponse>(
+                    request,
+                    Err(DebuggerError::Other(anyhow!(debug_error.clone()))),
+                )?;
+                Err(anyhow!(debug_error))
+            }
+        }
+    }
+
+    pub(crate) fn disconnect(
+        &mut self,
+        target_core: &mut CoreHandle,
+        request: Request,
+    ) -> Result<()> {
+        let arguments: DisconnectArguments = match get_arguments(&request) {
+            Ok(arguments) => arguments,
+            Err(error) => return self.send_response::<()>(request, Err(error)),
+        };
+
+        // We have a couple of drivers to determine the behaviour for this request.
+        let is_part_of_restart = arguments
+            .restart
+            .and_then(|is_part_of_restart| if is_part_of_restart { Some(()) } else { None })
+            .is_some();
+
+        // TODO: For now (until we do multicore), we will assume that both terminate and suspend translate to a halt of the core.
+        let must_halt_debuggee = arguments
+            .terminate_debuggee
+            .and_then(|terminate_debuggee| if terminate_debuggee { Some(()) } else { None })
+            .is_some()
+            || arguments
+                .suspend_debuggee
+                .and_then(|suspend_debuggee| if suspend_debuggee { Some(()) } else { None })
+                .is_some();
+
+        if !is_part_of_restart && must_halt_debuggee {
```

Comment:
I have had issues before where the reset timed out because of core activity, and since then, I default to making sure it is halted before I issue a reset. Is there a specific concern with forcing the halt a this point, or is this just an observation?